src/components/Card.jsx

```diff
@@ -1,30 +1,150 @@
-import React from 'react'
-import PropTypes from 'prop-types'
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Button, IconButton, Rating } from '@mui/material';
+import MoreVertIcon from '@mui/icons-material/MoreVert';
+import Menu from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
+import { ActionResultDialog } from './ActionResultDialog';
+import { ActionModal } from './ActionModal';
 
-const Card = props => {
-   const {title, genders, year, poster} = props;
+const Card = (props) => {
+  const { movie, handleEdit, handleDelete } = props;
+  const [showBtn, setShowBtn] = useState(false);
+  const [anchorEl, setAnchorEl] = useState(null);
+  const open = Boolean(anchorEl);
+  const [openDialog, setOpenDialog] = useState(false);
+  const [openModal, setOpenModal] = useState(false);
 
-   
+  const handleOpenModal = () => setOpenModal(true);
+  const handleCloseModal = () => setOpenModal(false);
+  const handleOpenDialog = () => setOpenDialog(true);
+  const handleCloseDialog = () => setOpenDialog(false);
+
+  const handleClick = (event) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+    setShowBtn(false);
+  };
+
+  const onDelete = () => {
+    handleDelete(movie.id);
+    handleClose();
+    handleCloseDialog();
+  };
 
   return (
-    <div className='card'>
-      <img src={poster} alt={title} className="card_poster"/>
-      <div className='card_top'>
-         <div className='card_title'>{title}</div>
-         <div className='card_year'>{year}</div>
+    <div className="card">
+      <div
+        style={{ backgroundImage: `url('${movie.poster}')` }}
+        className="card_poster"
+        onMouseEnter={() => setShowBtn(true)}
+        onMouseLeave={() => handleClose()}
+      >
+        {showBtn && (
+          <>
+            <IconButton
+              style={{ color: 'white' }}
+              className="cardButton"
+              size="small"
+              onClick={handleClick}
+            >
+              <MoreVertIcon />
+            </IconButton>
+            <Menu
+              id="basic-menu"
+              anchorEl={anchorEl}
+              open={open}
+              onClose={handleClose}
+              onClick={handleClose}
+              PaperProps={{
+                elevation: 0,
```

Comment:
I really don't like this inline css noise in .jsx files. Consider extracting styling into variables or styling components in a separate file and then importing them.